examples/sample-layers/label-layer/label-utils.js

```diff
@@ -21,48 +15,37 @@ function setTextStyle(ctx, fontSize) {
 export function makeTextureAtlasFromLabels(gl, {data, getLabel = x => x, fontSize = 48}) {
   const canvas = document.createElement('canvas');
   const ctx = canvas.getContext('2d');
-  setTextStyle(ctx, fontSize);
+  const textPainter = new TextAtlasPainter(ctx, MAX_CANVAS_WIDTH, fontSize);
+
+  textPainter.setTextStyle();
 
-  // measure texts
-  let row = 0;
-  let x = 0;
   const mapping = data.map((object, index) => {
     const string = getLabel(object);
-    const {width} = ctx.measureText(string);
-    if (x + width > MAX_CANVAS_WIDTH) {
-      x = 0;
-      row++;
-    }
+    const region = textPainter.getTextBounds(string);
+    // region =  {x, y, width, height}
+    // Anchor the label left aligned above the object, not centered.
+    // Anchor the height so bottom of label is above object.
     const iconMap = {
       object,
       index,
       string,
-      x,
-      y: row * fontSize,
-      width: Math.min(width, MAX_CANVAS_WIDTH),
-      height: fontSize,
-      mask: true
+      mask: true,
+      ...region,
+      anchorX: 0,
+      anchorY: region.height
     };
-    x += width;
     return iconMap;
   });
 
+  const extents = textPainter.getExtents();
   canvas.width = MAX_CANVAS_WIDTH;
-  canvas.height = (row + 1) * fontSize;
+  canvas.height = extents.height;
```

Comment:
`TextAtlasPainter` does not force dimensions to power of two. Does this still work with luma.gl 4?